tests/unit/test_reservation.py

```diff
@@ -208,5 +215,402 @@ def test___heterogenous_session_infos___create_sessions___grouped_by_instrument_
+
+    connections = reservation.get_connections(object, sites=0)
+
+    assert [conn.pin_or_relay_name for conn in connections] == [
+        "Pin1",
+        "Pin2",
+        "SystemPin1",
+        "SystemPin2",
+    ]
+    assert [conn.site for conn in connections] == [0, 0, -1, -1]
+    assert [conn.channel_name for conn in connections] == ["0", "1", "4", "5"]
+
+
+def test___system_pins___get_connection_with_system_pins_constant___only_system_pins_returned(
+    session_management_client: Mock,
+) -> None:
+    grpc_session_infos = _create_grpc_session_infos_with_system_pins()
+    reservation = MultiSessionReservation(session_management_client, grpc_session_infos)
+
+    connections = reservation.get_connections(object, sites=SITE_SYSTEM_PINS)
+
+    assert [conn.pin_or_relay_name for conn in connections] == ["SystemPin1", "SystemPin2"]
+    assert [conn.site for conn in connections] == [-1, -1]
+    assert [conn.channel_name for conn in connections] == ["4", "5"]
+
+
+def test___system_pins___get_connection_with_site_list___system_pins_returned_in_specified_order(
+    session_management_client: Mock,
+) -> None:
+    grpc_session_infos = _create_grpc_session_infos_with_system_pins()
+    reservation = MultiSessionReservation(session_management_client, grpc_session_infos)
+
+    connections = reservation.get_connections(object, sites=[0, SITE_SYSTEM_PINS, 1])
+
+    assert [conn.pin_or_relay_name for conn in connections] == [
+        "Pin1",
+        "Pin2",
+        "SystemPin1",
+        "SystemPin2",
+        "Pin1",
+        "Pin2",
+    ]
+    assert [conn.site for conn in connections] == [0, 0, -1, -1, 1, 1]
+    assert [conn.channel_name for conn in connections] == ["0", "1", "4", "5", "2", "3"]
+
+
 def _construct_session(session_info: SessionInformation) -> fake_driver.Session:
     return fake_driver.Session(session_info.resource_name)
+
+
+def _create_grpc_session_infos_for_ordering() -> (
+    List[session_management_service_pb2.SessionInformation]
+):
+    grpc_session_infos = create_nifoo_session_infos(4)
+    grpc_session_infos[0].channel_mappings.add(pin_or_relay_name="Pin1", site=0, channel="0")
+    grpc_session_infos[0].channel_mappings.add(pin_or_relay_name="Pin2", site=0, channel="1")
+    grpc_session_infos[1].channel_mappings.add(pin_or_relay_name="Pin3", site=0, channel="2")
+    grpc_session_infos[1].channel_mappings.add(pin_or_relay_name="Pin1", site=1, channel="3")
+    grpc_session_infos[2].channel_mappings.add(pin_or_relay_name="Pin2", site=1, channel="4")
+    grpc_session_infos[2].channel_mappings.add(pin_or_relay_name="Pin3", site=1, channel="5")
+    grpc_session_infos[3].channel_mappings.add(pin_or_relay_name="Pin4", site=0, channel="6")
+    grpc_session_infos[3].channel_mappings.add(pin_or_relay_name="Pin4", site=1, channel="7")
+    grpc_session_infos[3].instrument_type_id = "nibar"
+    return grpc_session_infos
+
+
+def _create_grpc_session_infos_with_system_pins() -> (
```

Comment:
@pbirkhol-ni @jasonmreding @jonathanmendez What test cases am I missing? Shared pins? Do we need to update the session management service for that?

Reply:
The LabVIEW implementation does not handle shared pins well. There is a follow up user story for how to handle shared pins with the following comment:

`
There isn't a great way to handle shared pins. If you have a shared pin across multiple sites, the channel mapping will only contain the shared pin on a single site. That means that if you call Get Connections on a single site, you may or may not get the shared pin. The workaround would be to always query all sites when trying to get a shared pin.
`

I think the solution may require an update to the session manager service since nothing in the session information currently indicates that a pin is shared.

Reply:
Moved to feature AB#2546523: Fully support querying connections for shared pins

BTW, I added an integration test for shared pins.